homeassistant/components/matter/climate.py

```diff
@@ -0,0 +1,291 @@
+                    self.target_temperature_high,
+                )
+        if command:
+            await self.matter_client.send_device_command(
+                node_id=self._endpoint.node.node_id,
+                endpoint_id=self._endpoint.endpoint_id,
+                command=command,
+            )
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set new target hvac mode."""
+        hvac_system_mode_map = {
+            HVACMode.HEAT: 4,
+            HVACMode.COOL: 3,
+            HVACMode.HEAT_COOL: 1,
+        }
+        system_mode_path = create_attribute_path_from_attribute(
+            endpoint_id=self._endpoint.endpoint_id,
+            attribute=clusters.Thermostat.Attributes.SystemMode,
+        )
+        system_mode_value = hvac_system_mode_map.get(hvac_mode)
+        if system_mode_value is None:
+            return
+        await self.matter_client.write_attribute(
+            node_id=self._endpoint.node.node_id,
+            attribute_path=system_mode_path,
+            value=system_mode_value,
+        )
+
+    @callback
+    def _update_from_device(self) -> None:
+        """Update from device."""
+        self._attr_current_temperature = self._get_temperature_in_degrees(
+            clusters.Thermostat.Attributes.LocalTemperature
+        )
+        # update hvac_mode from SystemMode
+        system_mode_value = int(
+            self.get_matter_attribute_value(clusters.Thermostat.Attributes.SystemMode)
+        )
+        match system_mode_value:
+            case SystemModeEnum.kAuto:
+                self._attr_hvac_mode = HVACMode.HEAT_COOL
+            case SystemModeEnum.kDry:
+                self._attr_hvac_mode = HVACMode.DRY
+            case SystemModeEnum.kFanOnly:
+                self._attr_hvac_mode = HVACMode.FAN_ONLY
+            case SystemModeEnum.kCool | SystemModeEnum.kPrecooling:
+                self._attr_hvac_mode = HVACMode.COOL
+            case SystemModeEnum.kHeat | SystemModeEnum.kEmergencyHeat:
+                self._attr_hvac_mode = HVACMode.HEAT
+            case _:
+                self._attr_hvac_mode = HVACMode.OFF
+        # running state is an optional attribute
+        # which we map to hvac_action if it exists (its value is not None)
+        self._attr_hvac_action = None
+        if running_state_value := self.get_matter_attribute_value(
+            clusters.Thermostat.Attributes.ThermostatRunningState
+        ):
+            match running_state_value:
+                case 1 | 8:
+                    self._attr_hvac_action = HVACAction.HEATING
+                case 2 | 16:
+                    self._attr_hvac_action = HVACAction.COOLING
+                case 4 | 32 | 64:
+                    self._attr_hvac_action = HVACAction.FAN
+                case _:
+                    self._attr_hvac_action = HVACAction.OFF
+        # update hvac_modes based on ControlSequenceOfOperation
+        control_sequence_value = int(
```

Comment:
Can we avoid updating hvac modes after setting up the entity or at least after setting hvac modes once?

Reply:
The available hvac_modes will depend on the available modes of the device if I have read the specs correctly. But @hidaris known way better than me as he has an actual device to test with.

If the modes are static, we could simply set it once, either in a "added_to_hass" callback or just by checking if the attribute has a value in the existing update callback

Reply:
Create an `__init__` method for the entity and set the static attributes.

Reply:
How is the __init__ of MatterEntity called? I couldn't find where it is

Reply:
I found it https://github.com/home-assistant/core/blob/8b159d0f479b889d82be8f88ded85f444a2fbc46/homeassistant/components/matter/adapter.py#L199